### PR TITLE
Setup refresh token

### PR DESCRIPTION
**Pull request by jreynard-code**
Dev Review rules:
- 3 PR max
- at the second feedback, prefer meet instead of github conversation

Definition of Done:
- [X] Testing
- [X] No error logs in web console (except iFrame errors)
- [ ] Create or Update documentation (README file for users/integrators)
- [X] Check the [test page](https://spaceport.cosmotech.com/confluence/pages/viewpage.action?spaceKey=QA&title=Azure+Sample+Webapp) in order to manually validate the feature 
- [X] Clean code (regroup configuration variable, typos, no console.log...)

_______________
Migrate from @cosmotech/api to @cosmotech/api-ts
Use axios lib for every API calls
Remove useless services
Adapt services used
Adapt sagas with new 'clientApi'
Add request interceptors for clientApi (adding Autorization header with refresh token )

### CODE REVIEW

File: src/services/dataset/DatasetService.js [hunk 1]
```diff
@@ -1,57 +1,25 @@
...
-function findAllDatasets (organizationId) {
-  return new Promise((resolve) => {
-    DatasetApi.findAllDatasets(organizationId, (error, data, response) => {
-      resolve({ error, data, response });
-    });
-  });
-}
+import { Api } from '../../configs/Api.config';
 
 function findDatasetById (organizationId, datasetId) {
```
Comment (csm-thu) [inline src/services/dataset/DatasetService.js:5]: Do we really need the functions `createDataset`, `updateDataset` and `deleteDataset` ? If these functions are only wrappers around the call to the API, we should be consistent between all services and create these wrappers for all calls to the API

File: src/services/Client.config.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Cosmo Tech.
```
Comment (csm-thu) [inline src/services/Client.config.js:1]: Can we discuss to find a better name than `Client.config.js` ? IMHO this is not a configuration file

File: src/services/Client.config.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import axios from 'axios';
+
+import { Auth } from '@cosmotech/core';
+
+const addInterceptors = (axiosInstance) => {
+  axiosInstance.interceptors.request.use(
+    async (request) => {
+      const tokens = await Auth.acquireTokens();
+      if (tokens !== undefined && tokens.accessToken !== undefined) {
```
Comment (csm-thu) [inline src/services/Client.config.js:12]: Could be simplified in `if (tokens?.accessToken)`

**Review by csm-thu — CHANGES_REQUESTED** [review]

File: src/services/Client.config.js (hunk 2 above)
Comment (jreynard-code) [reply src/services/Client.config.js:1]: ClientApi.js is ok for you? I named it like the constant that we export in it

Comment (csm-thu) [reply src/services/Client.config.js:1]: Yup, sounds good to me :+1:

File: src/services/dataset/DatasetService.js (hunk 1 above)
Comment (jreynard-code) [reply src/services/dataset/DatasetService.js:5]: By now, I let these "wrapper class" to not have a big breaking change between 1.3 and 1.4.
The ticket "Move to NPM Packages" (for v2) will normally fix this point.

File: src/utils/StorageUtils.js [hunk 4]
```diff
@@ -3,6 +3,35 @@
...
+      await Auth.isUserSignedIn();
+      // Get the new token
+      authAccessToken = localStorage.getItem('authAccessToken');
+      console.log('New token');
+      console.log(authAccessToken);
+    }
+  }
+  return authAccessToken;
+}; */
+
```
[the remark this answers is not in the document]
Comment (jreynard-code) [reply src/utils/StorageUtils.js:34]: I deleted the StorageUtils file, it's not used anymore ;)

**Pull request merged**